pkg/leanix/integration_hub.go

```diff
@@ -65,27 +65,35 @@ func SelfStartRun(fqdn string, accessToken string, datasource string) (*SelfStar
 	}
 	startResponse := SelfStartResponse{}
 	json.Unmarshal(responseData, &startResponse)
-	_, err = validateConnectorConfiguration(startResponse.ConnectorConfiguration)
-	if err != nil {
-		return nil, err
+	error := validateConnectorConfiguration(startResponse.ConnectorConfiguration)
+	if error != nil {
+		_, err = UpdateFailedProgressStatus(startResponse.ProgressCallbackUrl, "INVALID CONNECTOR CONFIGURATION: ABORTING IHUB RUN.")
+		if err != nil {
+			log.Errorf("Failed to update progress[%s] to Integration Hub", FAILED)
+		}
+		return &startResponse, error
 	}
+	log.Info("Connector Configuration is validated")
 	return &startResponse, nil
 }
 
-func validateConnectorConfiguration(configuration ConnectorConfiguration) (bool, error) {
+func validateConnectorConfiguration(configuration ConnectorConfiguration) error {
```

Comment:
Can the multiple `if` statements be converted to `if else` in order to process only 1 case at any time.

Reply:
I agree. Here it returns error with every `if`. It will not check the other if' condition as the 1st one returns an error. I will change it to `if..else` block, if it looks cleaner.

Reply:
What do you mean? The way it is right now, the function would finish once the first validation check fails. The other cases would not be checked since we return the errors right away

Reply:
Yes, @DennisBerthold-leanix.. exactly what I was trying to say.

Reply:
I mean every `if else` will have only `fmt.Error...`
final return will be the `fmt.Error..` and empty

Reply:
I think I missed clarifying it. thanks for the elaboration.